wpilibNewCommands/src/main/java/edu/wpi/first/wpilibj2/command/CommandScheduler.java

```diff
@@ -275,8 +280,8 @@ public void run() {
         for (Consumer<Command> action : m_interruptActions) {
           action.accept(command);
         }
+        m_scheduledCommands.remove(command);
         m_requirements.keySet().removeAll(command.getRequirements());
-        iterator.remove();
```

Comment:
Why is the order changing here? Not opposed to it, just trying to understand. 

Though I'm rather sure this will cause a CME: when iterating, you need to remove via the iterator.

Reply:
Consistency. When commands finish, they're descheduled, then requirements are freed up. Thought I'd do the same here (and yes, I do need to use the iterator).